reference/src/representation/pointers.md

```diff
@@ -0,0 +1,44 @@
+# Representation of reference and pointer types
+
+### Terminology
+
+Reference types are types of the form `&T` or `&mut T`.
+
+Raw pointer types are types of the form `*const T` or `*mut T`.
+
+### Representation
+
+The alignment of reference and raw pointer types is the word size.
+
+The sizes of `&T`, `&mut T`, `*const T` and `*mut T` are the same,
+and are at least one word.
+
+* If `T` is a trait, then the size of `&T` is two words.
+* If `T` is a sized type then the size of `&T` is one word.
+* The size of `&[T]` is two words.
+
+### Notes
+
+The representations of `&T` and `&mut T` are the same.
+
+The representation of `&T` when `T` is a trait is the same as that of:
+```rust
+#[repr(C)]
+struct DynObject {
+  data: &u8,
+  vtable: &usize,
```

Comment:
Why is this `vtable: &usize` ?

Reply:
To emphasize the fact that it's non-zero and word-aligned. Not part of the representation, but I believe we're going to be requiring this when we come to validity.

Reply:
This seems like a very confusing way to make that point, and since it's not even really part of what this write-up is supposed to be about at this point, nor particularly useful, I'd really prefer not doing that. Just `*const ()` is good enough for now.

Reply:
I made them both `&u8`.